Add preg_quote to except and only filters

pqr opened the pull request:
Previous version did not quoted except and only filters in ShouldCollect, so following example did not preg_match because question symbol is special: `index.php?some_param`.

This commit add preg_quote($pattern) to properly escape special regex symbols in the pattern.

mahagr commented on Clockwork/Request/ShouldCollect.php:
```diff
@@ -78,7 +78,7 @@ protected function passExcept(IncomingRequest $request)
 
 		foreach ($this->except as $pattern) {
-			if (preg_match('#' . str_replace('#', '\#', $pattern) . '#', $request->uri)) return false;
+			if (preg_match('#' . preg_quote($pattern) . '#', $request->uri)) return false;
```
Pass delimiter: `preg_quote($pattern, '#')`, though in this case # will automatically be quoted anyway.

pqr replied:
In my case I want to quote a question symbol: '?'

Actual use case: I have 'api' config option:
`'api' => 'index.php?a=Clockwork&request='`

Then in source code I that api url automatically ignored - this is cool, but does not work!
https://github.com/itsgoingd/clockwork/blob/543de6b1c17021183be5fd2dd6c7ffa153d9ab2d/Clockwork/Support/Vanilla/Clockwork.php#L242
```
// don't collect data for Clockwork requests
$this->clockwork->shouldCollect()->except(rtrim($this->config['api'], '/'));
```

Why it does not work?
Because `preg_match('#index.php?a=Clockwork&request=#', $request->uri)` returns 0 even when `$request->uri === '/index.php?a=Clockwork&request=12345'`.

So we need to escape **?** symbol (and many others special regex symbols) in the first argument of `preg_match`

mahagr replied:
https://www.php.net/manual/en/function.preg-match.php

According to the doc, the second parameter should have the pattern delimiter, which is '#' as `preg_match('#XXX#')` starts and ends with #. Yes, in this case it can be left out, but I usually consider missing delimiter being a bug when I quickly browse through the code.

itsgoingd commented:
Hey, being able to pass regexes to except is an intentional feature. Otherwise we would not be able to specify non-trivial filters.

The real issue here seems to be not running the api path through `preg_quote` *before* passing it to except. Can you please update your PR?

Btw. would love to hear about your experience using Clockwork in a vanilla php app.

pqr commented:
@itsgoingd yes, you are right about `preg_quote` before passing it to except, I've updated the Pull Request.

The whole experience with valiant php app:
good:
- I was able to quickly add Clockwork to a vanila project (within few hours)
- It works like a charm, see a big value-added

bad:
- lack of documentation about all available methods. I researched the source code of Clockwork to find out what is possible and how to do it. Especially Laravel integration was helpful as an example
- no autocomplete in PhpStorm on `Clockwork\Support\Vanilla\Clockwork` instance because under the hood it proxies all method calls via `__call` and `__callStatic` to the `Clockwork\Clockwork as BaseClockwork` instance. May be it would be helpful to annotate `Vanilla\Clockwork` class with PHPDoc `@method` comments
- as a result, I introduced my own `My\Clockwork` class with explicit methods definitions and used it in the project with autocomplete and other useful PhpStorm inspections

mahagr commented:
I was just talking about lack of the phpdocs in the project yesterday (in Discord). I agree that the code should provide phpdocs to provide self-documentation, possibility to run static analyzer tools (such as phpstan) and to allow IDE to autocomplete the code.

itsgoingd commented:
Yeah, both of these issues are something I'm aware of and want to improve. Thanks for the PR and feedback!
